### PR TITLE
34-fix-dimension-relabelingmisorded dims

### PR DESCRIPTION
**Pull request by jonaswagner2826**
The primary update is changing linMap(), copy(), and transform() to allow inDims as input and have a warning if not provided (although removal of no inDims would also be a good idea but left for backwards compatibility)

There is also a supportFunc method for memZono that will call the underlying supportFunc() for appropriete dimensions.

### CODE REVIEW

File: @memZono/memZono.m [hunk 1]
```diff
@@ -376,9 +411,33 @@
...
+            %     outDims = varargin{1}; 
+            %     warning('lack of inDims specification can cause issues with dimension ordering'); 
+            % else
+            %     inDims = varargin{1};
+            %     outDims = varargin{2};
+            % end
+            if ~iscell(inDims); inDims = memZono.genKeys(inDims,1:size(M,1)); end
             if ~iscell(outDims); outDims = memZono.genKeys(outDims,1:size(M,1)); end
-            out = in.transform([],M,[],outDims);
+            out = in.transform([],M,inDims,outDims).projection(outDims);
```
Comment (jruths) [inline @memZono/memZono.m:425]: I'm not sure we need the projection after the call to transform....?

File: @abstractZono/abstractZono.m [hunk 2]
```diff
@@ -77,6 +71,16 @@
...
+        function [lb,ub] = bounds(obj)
+            bb = obj.boundingBox;
+            ub = bb.c + sum(abs(bb.G),2);
+            lb = bb.c - sum(abs(bb.G),2);
+        end
+    end
+
+    %% Additional Overide Methods
+    methods
+        function out = linMap(in,M); out = mtimes(M,in); end
```
Comment (jruths) [inline @abstractZono/abstractZono.m:83]: How does this syntax even work?  This defines linMap and mtimes?  as the same thing?  Are these changes needed?  I'd like to get buy in from Dr. Koeln if we are going to make changes to the main library....

File: docs/refs/Althoff2010.pdf
Comment (jruths) [inline docs/refs/Althoff2010.pdf:1]: Is there a precedent for including pdfs in the zonoLAB repository?

File: @abstractZono/abstractZono.m (hunk 2 above)
Comment (jonaswagner2826) [reply @abstractZono/abstractZono.m:83]: All this does is switch the order calling mtimes() so that it will call the correct method (needed for optimvar type stuff).

File: @memZono/memZono.m (hunk 1 above)
Comment (jonaswagner2826) [reply @memZono/memZono.m:425]: I'm not sure if it should be needed (that's a further disscussion) but as it is now the non-specified dimensions are just passed through the transformation, thus a projection is required to only output the appropriete dimensions for linMap() if the original set has extra dimensions that aren't needed for output

**Review by jonaswagner2826 — COMMENTED** [review]
> Is there a precedent for including pdfs in the zonoLAB repository? 

Technically yes. The pdf's (and ultimently all the notes I've had in docs) have been included in the upsteam version of zonoLAB for awhile. It was never disscussed though.

File: docs/refs/Althoff2010.pdf
Comment (jonaswagner2826) [reply docs/refs/Althoff2010.pdf:1]: Technically yes. The pdf's (and ultimently all the notes I've had in docs) have been included in the upsteam version of zonoLAB for awhile. It was never disscussed though.

**Review by jonaswagner2826 — COMMENTED** [review]
> I'm not sure we need the projection after the call to transform....? 

@jruths - I've updated the transform() w/ named-value arguments to allow for modified functionality of the passDims functionality as you previously alluded to. Please see if this is prefered or if the projection() method I had used is better and we can revert it.
